lib/usd/translators/meshWriter.cpp

```diff
@@ -180,7 +511,61 @@ bool PxrUsdTranslators_MeshWriter::writeMeshAttrs(
 
     // Set mesh attrs ==========
     // Write points
-    UsdMayaMeshWriteUtils::writePointsData(geomMesh, primSchema, usdTime, _GetSparseValueWriter());
+    /*
+     NOTE: (yliangsiew) Because we cannot assume that the first frame of export
+     will have no blendshape targets activated, and we want to write out the
+     points _without_ the influence of any blendshapes, (or any other deformers,
+     for that matter; just that we haven't implemented support for other
+     deformers yet, so until we do that, we can leave the effect of other
+     deformers "baked" into the base/"pref" pose) we need to deactivate all the
+     blendshape targets here _before_ writing out the data.
+    */
```

Comment:
I think you can achieve this in a simpler way by reading MFnGeometryFilter::getInputGeometry(), or directly reading the inputGeometry attribute of the blend shape. That will give you the input mesh without any of the blend shape's modifications. And it will leave any any deformations which occur after. We could have something like blendShape->skinCluster->Wrap->mesh. In the code that is here you'd end up with just the wrap applied, and then using that as the input to the blendShape isn't correct.

The inputGeometry could also be animated. I see some checks above for dealing with time but I'm not sure they capture this fact.

Reply:
This might be a better approach to finding the "pref" pose and I can give this a
shot...though I'm a little concerned about just how much I need to support here,
especially when other deformers get thrown into the mix in the future.